### PR TITLE
fixes for autocomplete implementation

### PR DESCRIPTION
**Pull request by kurund**
## Summary

* Autocomplete field generates several warnings although it's working. This PR attempts to clean up implementation and the places where it is used.

## Test Plan

* Included in the PR

### CODE REVIEW

File: src/containers/InteractiveMessage/InteractiveOptions/InteractiveOptions.tsx [hunk 1]
```diff
@@ -62,6 +62,7 @@ export const InteractiveOptions: React.SFC<InteractiveOptionsProps> = ({
     if (templateType === LIST) {
       template = (
         <ListReplyTemplate
+          key={index}
```
Comment (shreesh1993) [inline src/containers/InteractiveMessage/InteractiveOptions/InteractiveOptions.tsx:65]: Is it okay to use the index as a key? or should we go for just a random string such as the current date? Sometime back when I used index as key, deepscan gave me an error :(

File: src/containers/StaffManagement/StaffManagment.test.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ test('it should have a help link', async () => {
   // loading is show initially
   expect(screen.getByText('Loading...')).toBeInTheDocument();
   await waitFor(async () => await new Promise((resolve) => setTimeout(resolve, 0)));
-  const helpButton = screen.getByTestId('helpButton');
+  const helpButton = await screen.getByTestId('helpButton');
```
Comment (shreesh1993) [inline src/containers/StaffManagement/StaffManagment.test.tsx:33]: I've updated this test case, wrapped code under `await waitFor` in the Shamoon branch.
note: **await** is used with **findBy**

**Review by shreesh1993 — APPROVED** [review]

File: src/containers/InteractiveMessage/InteractiveOptions/InteractiveOptions.tsx (hunk 1 above)
Comment (kurund) [reply src/containers/InteractiveMessage/InteractiveOptions/InteractiveOptions.tsx]: Nope, we should not be directly using the index of the map for the key. I have updated the code.

File: src/containers/StaffManagement/StaffManagment.test.tsx (hunk 2 above)
Comment (kurund) [reply src/containers/StaffManagement/StaffManagment.test.tsx:33]: I don't see any open PR's., what branch?

> note: await is used with findBy

Not sure what you mean here. Let's discuss tomorrow.

Comment (kurund) [reply src/containers/StaffManagement/StaffManagment.test.tsx]: Fixed.

**Comment by kurund** [comment]
@shreesh1993 

Merging this as per our discussion.

**Pull request merged**